helper/krylovHelper.cpp

```diff
@@ -0,0 +1,160 @@
+//--------------------------------------------------------------------------
+//Class Parameter
+
+
+std::string parameter::getName()
+{
+    return p_name;
+}
+
+
+//--------------------------------------------------------------------------
+//Class krylovHelper
+
+krylovHelper::krylovHelper() {}
+
+void krylovHelper::timeEvolve() {}
+
+void krylovHelper::saveResult() {}
+
+//--------------------------------------------------------------------------
+//Class HDF5Helper
+
+void HDF5Helper::saveResult()
+{
+
+    std::string outputFileName = fileName;
+
+    parameter_list::iterator paraIter;
+    observable_list::iterator obsIter;
+
+    //Create string with parameter and its value
+    for (paraIter = para_list.begin(); paraIter != para_list.end(); paraIter++)
+        outputFileName += "_" + (*paraIter)->getName() + (*paraIter)->getData();
+
+    std::cout << outputFileName << std::endl;
+
+    for (obsIter = obs_list.begin(); obsIter != obs_list.end(); obsIter++)
+        std::cout << (*obsIter)->getName() << std::endl;
+
+
+
+    //Sort data from time ordering to observable ordering
+    double** nicelySorted = new double* [nbObservables];
+    for (int i = 0; i < nbObservables; ++i)
+        nicelySorted[i] = new double[results->nSamples];
+
+    for (int j = 0; j < nbObservables; j++)
+    {
+        for (unsigned int i = 0; i < results->nSamples; i++)
+        {
+            nicelySorted[j][i] = (results->sampling->values + nbObservables * i + j)->real();
+        }
+    }
+
+
+    //Use HDF output if HDF5 libraries are discovered during compiling
+#ifdef USE_HDF
+
+    outputFileName += ".h5";
+    H5File fileHh(outputFileName.c_str(), H5F_ACC_TRUNC);
+
+    for (int j = 0; j < nbObservables; j++)
+    {
+        int NX = results->nSamples;
+        const int RANK = 1;
+        hsize_t dimsf[RANK];
+        dimsf[0] = NX;
+        DataSpace dataspace(RANK, dimsf);
+        FloatType datatype(PredType::NATIVE_DOUBLE);
+        datatype.setOrder(H5T_ORDER_LE);
+        std::string modeNumber = "mode" + std::to_string(j);
+        dataset = fileHh.createDataSet(modeNumber.c_str(), datatype, dataspace);
+        dataset.write(nicelySorted[j], PredType::NATIVE_DOUBLE);
+
+        writeAttributes();
+
+        dataset.close();
+
+    }
+    //If not write data to simple csv files
+#else
+
+    for (int j = 0; j != nbObservables; j++)
+    {
+        std::string fileNameCSV = outputFileName + "mode" + std::to_string(j) + ".csv";
+        std::ofstream outputfile;
+        outputfile.open(fileNameCSV);
+        for (int i = 0; i != (results->nSamples) - 1; i++)
+            outputfile << nicelySorted[j][i] << ", ";
+        outputfile << nicelySorted[j][(results->nSamples) - 1];
+        outputfile.close();
+    }
+
+#endif
+    //end write obserables to file
+
+
+    //clean up
+    for (int i = 0; i < nbObservables; i++)
+    {
+        delete[] nicelySorted[i];
+    }
+    delete[] nicelySorted; 
+
+
+}
+
+void HDF5Helper::writeAttributes() 
```

Comment:
Shouldn't this only be defined if hdf5 was found (i.e. inside #ifdef)?

Reply:
Yes! This is an error and prevents compilation in case HDF is not found.